warehouse/query-core/src/main/java/datawave/query/iterator/ivarator/IvaratorCacheDirConfig.java

```diff
@@ -0,0 +1,138 @@
+        this(basePathURI, priority, minAvailableStorageMB, DEFAULT_MIN_AVAILABLE_STORAGE_PERCENT);
+    }
+    
+    public IvaratorCacheDirConfig(String basePathURI, int priority, double minAvailableStoragePercent) {
+        this(basePathURI, priority, DEFAULT_MIN_AVAILABLE_STORAGE_MB, minAvailableStoragePercent);
+    }
+    
+    private IvaratorCacheDirConfig(String basePathURI, int priority, long minAvailableStorageMB, double minAvailableStoragePercent) {
+        this.basePathURI = basePathURI;
+        this.priority = priority;
+        this.minAvailableStorageMB = minAvailableStorageMB;
+        this.minAvailableStoragePercent = minAvailableStoragePercent;
+    }
+    
+    @JsonIgnore
+    public boolean isValid() {
+        boolean result = true;
+        
+        if (!basePathURI.startsWith("file:/") && !basePathURI.startsWith("hdfs:/")) {
+            log.warn("Invalid basePathURI for IvaratorCacheDirConfig.  'basePathURI' scheme must be either 'file:' or 'hdfs:'");
+            result = false;
+        }
+        
+        if (minAvailableStorageMB < 0l) {
+            log.warn("Invalid minAvailableStorageMB for IvaratorCacheDirConfig.  'minAvailableStorageMB' must be greater than or equal to 0");
+            result = false;
+        }
+        
+        if (minAvailableStoragePercent < 0.0 || minAvailableStoragePercent > 1.0) {
+            log.warn("Invalid minAvailableStoragePercent for IvaratorCacheDirConfig.  'minAvailableStoragePercent' must be between 0.0 and 1.0 inclusive");
+            result = false;
+        }
+        
+        return result;
+    }
+    
+    public String getBasePathURI() {
+        return basePathURI;
+    }
+    
+    public int getPriority() {
+        return priority;
+    }
+    
+    public long getMinAvailableStorageMB() {
+        return minAvailableStorageMB;
+    }
+    
+    public double getMinAvailableStoragePercent() {
+        return minAvailableStoragePercent;
+    }
+    
+    public static String toJson(IvaratorCacheDirConfig ivaratorCacheDirConfig) throws JsonProcessingException {
+        return toJson(Collections.singletonList(ivaratorCacheDirConfig));
+    }
+    
+    public static String toJson(List<IvaratorCacheDirConfig> ivaratorCacheDirConfigs) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(SerializationFeature.WRITE_SINGLE_ELEM_ARRAYS_UNWRAPPED, true);
+        return objectMapper.writeValueAsString(ivaratorCacheDirConfigs);
+    }
+    
+    public static List<IvaratorCacheDirConfig> fromJson(String jsonArray) throws JsonProcessingException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        objectMapper.configure(DeserializationFeature.ACCEPT_SINGLE_VALUE_AS_ARRAY, true);
+        return new ArrayList<>(Arrays.asList(objectMapper.readValue(jsonArray, IvaratorCacheDirConfig[].class)));
+    }
```

Comment:
Could leverage a single static final ObjectMapper instead, and just configure once via static block

Reply:
I thought about that.  Is ObjectMapper thread safe though?  Does it even need to be the way I'm using it...  Let me think on this.

Reply:
Yep it's threadsafe, as long as you don't modify it's config after the fact

Reply:
> Yep it's threadsafe, as long as you don't modify it's config after the fact

i.e., as long as different threads don't need to alter the config for whatever reason

Reply:
Got it!  pr updated.